src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobConfigurationValidator.cs

```diff
@@ -0,0 +1,76 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Net;
+using System.Text;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Features.Operations.Export.AccessTokenProvider;
+using Microsoft.Health.Fhir.Core.Features.Operations.Export.ExportDestinationClient;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.Export
+{
+    public class ExportJobConfigurationValidator : IExportJobConfigurationValidator
+    {
+        private readonly ExportJobConfiguration _exportJobConfiguration;
+        private readonly IExportDestinationClientFactory _exportDestinationClientFactory;
+        private readonly IAccessTokenProviderFactory _accessTokenProviderFactory;
+        private readonly ILogger<ExportJobConfigurationValidator> _logger;
+
+        public ExportJobConfigurationValidator(
+            IOptions<ExportJobConfiguration> exportJobConfiguration,
+            IExportDestinationClientFactory exportDestinationClientFactory,
+            IAccessTokenProviderFactory accessTokenProviderFactory,
+            ILogger<ExportJobConfigurationValidator> logger)
+        {
+            EnsureArg.IsNotNull(exportJobConfiguration?.Value, nameof(exportJobConfiguration));
+            EnsureArg.IsNotNull(exportDestinationClientFactory, nameof(exportDestinationClientFactory));
+            EnsureArg.IsNotNull(accessTokenProviderFactory, nameof(accessTokenProviderFactory));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _exportJobConfiguration = exportJobConfiguration.Value;
+            _exportDestinationClientFactory = exportDestinationClientFactory;
+            _accessTokenProviderFactory = accessTokenProviderFactory;
+            _logger = logger;
+        }
+
+        public bool ValidateExportJobConfig()
+        {
+            if (string.IsNullOrWhiteSpace(_exportJobConfiguration.StorageAccountType) ||
+                !_exportDestinationClientFactory.IsSupportedDestinationType(_exportJobConfiguration.StorageAccountType))
+            {
+                throw new ExportJobConfigValidationException(string.Format(Resources.UnsupportedDestinationType, _exportJobConfiguration.StorageAccountType), HttpStatusCode.BadRequest);
+            }
+
+            // Check whether the config contains a uri to a storage account or a connection string.
+            if (Uri.TryCreate(_exportJobConfiguration.StorageAccountConnection, UriKind.Absolute, out Uri resultUri))
```

Comment:
You already have an interface for handling acess token (i would make it more generic so it can handle how authentication is handled) and then just separate the connection to another implementation of the interface rather than depending on whther the conection is a "url" or not.

Reply:
Refactored the code to handle authentication in a separate interface/class and removed the `ExportJobConfigurationValidator`